lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,461 @@
+            return []
+        }
+
+        // Validate the body of the description function.
+        //   1. The code block must have a single item
+        //   2. The single item must be a return of a string literal
+        //   3. Later on, the interpolation in the string literal will be validated.
+        guard let codeBlock = onlyBinding.accessorBlock?.accessors.as(CodeBlockItemListSyntax.self),
+              let descriptionString = codeBlock.onlyReturnExpr?.as(StringLiteralExprSyntax.self) else {
+            let message: ErrorMessage = "body must consist of a single string literal"
+            context.diagnose(node: declaration, error: message)
+            return []
+        }
+
+        // Iterate the string's segments, and convert property expressions into LLDB variable references.
+        var summarySegments: [String] = []
+        for segment in descriptionString.segments {
+            switch segment {
+            case let .stringSegment(segment):
+                summarySegments.append(segment.content.text)
+            case let .expressionSegment(segment):
+                guard let onlyLabeledExpr = segment.expressions.only, onlyLabeledExpr.label == nil else {
+                    // This catches `appendInterpolation` overrides.
+                    let message: ErrorMessage = "unsupported custom string interpolation expression"
+                    context.diagnose(node: segment, error: message)
+                    return []
+                }
+
+                let expr = onlyLabeledExpr.expression
+
+                // "Parse" the expression into a flattened chain of property accesses.
+                var propertyChain: [DeclReferenceExprSyntax] = []
+                var invalidExpr: ExprSyntax? = nil
+                if let declRef = expr.as(DeclReferenceExprSyntax.self) {
+                    // A reference to a single property on self.
+                    propertyChain.append(declRef)
+                } else if let memberAccess = expr.as(MemberAccessExprSyntax.self) {
+                    do {
+                        propertyChain = try memberAccess.propertyChain()
+                    } catch let error as UnexpectedExpr {
+                        invalidExpr = error.expr
+                    }
+                } else {
+                    // The expression was neither a DeclReference nor a MemberAccess.
+                    invalidExpr = expr
+                }
+
+                if let invalidExpr {
+                    let message: ErrorMessage = "only references to stored properties are allowed"
+                    context.diagnose(node: invalidExpr, error: message)
+                    return []
+                }
+
+                // Eliminate explicit self references. The debugger doesn't support `self` in
+                // variable paths.
+                propertyChain.removeAll(where: { $0.baseName.text == "self" })
```

Comment:
I think you should be able to check for the token kind instead of the text here

```suggestion
                propertyChain.removeAll(where: { $0.baseName.tokenKind == .self })
```

Reply:
nice

Reply:
seems I need:

```
propertyChain.removeAll(where: { $0.baseName.tokenKind == .keyword(.self) })
```

Reply:
Oh, yes. I wrote it in GitHub, so didn’t remember that you needed to add `.keyword`.